### PR TITLE
Adding a link to the general terms and conditions in the checkout footer (Opquast n°45)

### PR DESCRIPTION
**Pull request by Sinepel**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop 
| Description?  | Currently the general terms and conditions of sale are not accessible in the order tunnel except at step 4 / the payment step. In order to comply with the OPQUAST checklist and recommendation number 45, the general terms and conditions of sale should be accessible from all pages of the site. I have therefore added a link to this page in the footer of the order tunnel. This link opens a modal in order not to leave the tunnel.
| Type?         | improvement 
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | Add a product to your cart & go to the checkout process

Capture d’écran 2020-08-21 à 15 13 16

:notebook: BC Break
* The modal `<div class="modal fade" id="modal">` has been moved from `themes/classic/templates/checkout/_partials/steps/payment.tpl` (block `step_content`) to `themes/classic/templates/checkout/_partials/footer.tpl` (block `footer`).

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20203)

### CODE REVIEW

**Comment by matks** [comment]
Oh, I see you are a Opquast expert 🧐 😄

**Comment by Sinepel** [comment]
Yes, I have potentially more to come !

**Review by matks — DISMISSED** [review]
Hi ! I think the PR and the provided explanations are good.

Maybe the only thing I am not 100% is where to put this link. If I am not wrong, you added the link at the same place as the "Copyright 2020 - Ecommerce software by PrestaShop".

Consequently I ping the Product and Design teams to discuss with them where would be the best place for this link.

**Comment by matks** [comment]
ping @PrestaShop/prestashop-product-team @TristanLDD see above the design question

@Sinepel Would it be possible to provide a screenshot of how it looks like 😊  ?

**Comment by Sinepel** [comment]
I understand :) !

I have modified the description of the PR to include the screenshot.

File: controllers/front/OrderController.php [hunk 1]
```diff
@@ -356,4 +358,25 @@ public function displayAjaxAddressForm()
...
     }
+
+    /**
+     * Return default TOS link for checkout footer
+     *
+     * @return string
+     */
+    protected function getDefaultTermsAndConditions()
+    {
+        $cms = new CMS((int) Configuration::get('PS_CONDITIONS_CMS_ID'), $this->context->language->id);
```
Comment (PierreRambaud) [inline controllers/front/OrderController.php:320]: What if there is no default CMS page loaded? :)

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I think some checks must be added in `getDefaultTermsAndConditions` to be sure the page exists

**Comment by Sinepel** [comment]
> I think some checks must be added in `getDefaultTermsAndConditions` to be sure the page exists

@PierreRambaud Is that okay ? 
I saw that the PS_CONDITIONS_CMS_ID configuration value was set to 3 by default and that it was impossible not to choose anything for this field in BO.

Thanks ! :)

File: controllers/front/OrderController.php (hunk 1 above)
Comment (Sinepel) [reply controllers/front/OrderController.php:320]: @PierreRambaud it's OK with my last check ? Thanks !

File: controllers/front/OrderController.php [hunk 2]
```diff
@@ -356,4 +358,29 @@ public function displayAjaxAddressForm()
...
+     * @return string|bool
+     */
+    protected function getDefaultTermsAndConditions()
+    {
+        $cms = new CMS((int) Configuration::get('PS_CONDITIONS_CMS_ID'), $this->context->language->id);
+        $link = $this->context->link->getCMSLink($cms, $cms->link_rewrite, (bool) Configuration::get('PS_SSL_ENABLED'));
+
+        if (!$cms->id || !Validate::isLoadedObject($cms)) {
+            return false;
+        }
```
Comment (Progi1984) [inline controllers/front/OrderController.php]:
```suggestion
        if (!$cms->id || !Validate::isLoadedObject($cms)) {
            return false;
        }
        
        $link = $this->context->link->getCMSLink($cms, $cms->link_rewrite, (bool) Configuration::get('PS_SSL_ENABLED'));
```

**Review by Progi1984 — COMMENTED** [review]
A small change ;) and it's :ok_hand: for me.

**Review by Progi1984 — DISMISSED** [review]

**Comment by Sinepel** [comment]
> A small change ;) and it's 👌 for me.

It makes more sense indeed, thank you! :)

File: controllers/front/OrderController.php [hunk 3]
```diff
@@ -356,4 +359,30 @@ public function displayAjaxAddressForm()
...
+    /**
+     * Return default TOS link for checkout footer
+     *
+     * @return string|bool
+     */
+    protected function getDefaultTermsAndConditions()
+    {
+        $cms = new CMS((int) Configuration::get('PS_CONDITIONS_CMS_ID'), $this->context->language->id);
+
+        if (!$cms->id || !Validate::isLoadedObject($cms)) {
```
Comment (PierreRambaud) [inline controllers/front/OrderController.php]:
```suggestion
        if (!Validate::isLoadedObject($cms)) {
```

The first check is already done in `isLoadedObject` :sweat_smile:

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Last one thing and it's ok for me. Sorry :sweat_smile:

File: controllers/front/OrderController.php (hunk 3 above)
Comment (Sinepel) [reply controllers/front/OrderController.php]: Exat! I had only added the first one at the beginning and added the `isLoadedObject` then, I forgot to remove it! Thank you! :)

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by MatShir** [comment]
Great idea! @TristanLDD @prestascott What are your thoughts?

**Comment by prestascott** [comment]
Hi everyone,

Thank you for your report @Sinepel. 
I agree with you, this is a nice improvement since it's now a standard for websites to have 'Terms and Conditions' link in the footer.

However, i'm not sure about putting the link next to the Copyright because it's not really related. Imo, we should separate them and put the link above/below to the Copyright.

**Comment by matks** [comment]
> Hi everyone,
> 
> Thank you for your report @Sinepel.
> I agree with you, this is a nice improvement since it's now a standard for websites to have 'Terms and Conditions' link in the footer.
> 
> However, i'm not sure about putting the link next to the Copyright because it's not really related. Imo, we should separate them and put the link above/below to the Copyright.

@Sinepel Will you update the PR to follow @prestascott feedback ? 😊

**Comment by Sinepel** [comment]
@matks Sure, I do that :)

**Comment by Sinepel** [comment]
@matks It's OK and I resolved the conflict

**Review by matks — APPROVED** [review]
@Sinepel Just checked the PR, everything looks good. So I approve the PR. I also updated your screenshot.

There is however something that puzzles me 🤔  : moving`<div class="modal fade" id="modal">` . I am wondering if this could introduce regressions.

**Comment by matks** [comment]
> There is however something that puzzles me 🤔 : moving`<div class="modal fade" id="modal">` . I am wondering if this could introduce regressions.

Wdyt @NeOMakinG ?

**Comment by NeOMakinG** [comment]
I would say that this is a BC, even if I doubt a lot of peoples changed something about this particular part !

@matks

**Comment by Sinepel** [comment]
@matks @NeOMakinG 
Personally, since we have been working on the 1.7, for 2/3 years already, we have never touched this part.

**Comment by NeOMakinG** [comment]
I would agree with accepting it and just tag it as a BC and document it, just in case!

**Comment by Progi1984** [comment]
@NeOMakinG Labeled it as a BC and BC Break documented

**Review by Progi1984 — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM thanks @Sinepel !!

**Comment by PierreRambaud** [comment]
Thanks @Sinepel

**Pull request merged**